src/hooks/useSuspendWhen.tsx

```diff
@@ -0,0 +1,15 @@
+import { useState } from 'react'
+
+export const useSuspendWhen = (shouldSuspend = false) => {
```

Comment:
then when matches the `<Suspend when={}>`, which I've always like when for some reason, maybe just because if is a reserved word and sometimes it causes issues when you want to use a variable and then pass it to an object without having to rename it, but i can change both to if in this case